src/app/_utils/Auth.ts

```diff
@@ -1,7 +1,9 @@
-import { CanActivate } from "@angular/router";
+import { CanActivateChild, CanLoad } from '@angular/router';
 
-export default class Auth implements CanActivate {
-    canActivate(): boolean {
+export default class Auth implements CanActivateChild, CanLoad {
+    canLoad = () => this.canActivateChild(); //in this case it is the same auth, but it could be different
```

Comment:
👍

Reply:
If we would scale this solution, that would be much better to move functionality to a separate method and reuse it in canLoad and in canActivateChild.
This way it will be much more easy to test and refactor